lib/sock_js/sock_js_parser.dart

```diff
@@ -0,0 +1,100 @@
+import 'dart:convert';
+import 'dart:typed_data';
+import 'package:stomp_dart_client/parser.dart';
+import 'package:stomp_dart_client/stomp_frame.dart';
+import 'package:stomp_dart_client/stomp_parser.dart';
+
+class SockJSParser implements Parser {
+
+  StompParser _stompParser;
+
+  SockJSParser(Function(StompFrame) onStompFrame, [Function onPingFrame]){
+    _stompParser = StompParser(onStompFrame, onPingFrame);
+  }
+
+  @override
+  void parseData(dynamic data) {
+    Uint8List byteList;
+    if (data is String) {
+      byteList = Uint8List.fromList(utf8.encode(data));
+    } else if (data is List<int>) {
+      byteList = Uint8List.fromList(data);
+    } else {
+      throw UnsupportedError('Input data type unsupported');
+    }
+
+    _collectData(byteList);
+  }
+
+  void _collectData(Uint8List byteList) {
+    if (byteList == null || byteList.isEmpty) {
+      return;
+    }
+
+    var msg = utf8.decode(byteList);
+    var type = msg.substring(0, 1);
+    var content = msg.substring(1);
+
+    // first check for messages that don't need a payload
+    switch (type)
+    {
+      case 'o': // Open frame
+      case 'h': // Heartbeat frame
+          return;
+      default:
+        break;
+    }
+
+    if (content.isEmpty){
+      return;
+    }
+
+    dynamic payload;
+    try
+    {
+      payload = json.decode(content);
+    }
+    catch(exception){
+      return;
+    }
+
+    switch (type)
+    {
+      case 'a'://Array of messages
+        if (payload is List)
+        {
+          for (var item in payload)
+          {
+            _stompParser.parseData(item);
+          };
+        }
+        break;
+      case 'm'://message
+        _stompParser.parseData(payload);
+        break;
+      case 'c'://Close frame
+        if(_stompParser.onStompFrame != null){
+          _stompParser.onStompFrame(StompFrame(command: 'ERROR'));
+        }
```

Comment:
I think this should probably call _onDone in the handler so that the client closes the sink and also starts reconnect attempts if this i an unexpected close frame. As i understood, this is not an error on the Stomp protocol level.
The callback could be passed as a parameter to the constructor (like onPingFrame)